{"text": "2024-01-23 06:29:03 | INFO    | __main__:stash:131 | run - {'image': '20231121', 'version': '0.10.1.post2', 'role': 'EpicPlayer', 'headless': True}\n", "record": {"elapsed": {"repr": "0:00:01.011720", "seconds": 1.01172}, "exception": null, "extra": {"image": "20231121", "version": "0.10.1.post2", "role": "EpicPlayer", "headless": true}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 131, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3635, "name": "MainProcess"}, "thread": {"id": 140311102258048, "name": "MainThread"}, "time": {"repr": "2024-01-23 06:29:03.825124+08:00", "timestamp": 1705962543.825124}}}
{"text": "2024-01-23 06:29:10 | WARNING    | epic_games.agent:get_order_history:467 | Failed to get order history, cookie may have expired - {}\n", "record": {"elapsed": {"repr": "0:00:07.953219", "seconds": 7.953219}, "exception": null, "extra": {}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/epic_games/agent.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 467, "message": "Failed to get order history, cookie may have expired", "module": "agent", "name": "epic_games.agent", "process": {"id": 3635, "name": "MainProcess"}, "thread": {"id": 140311102258048, "name": "MainThread"}, "time": {"repr": "2024-01-23 06:29:10.766623+08:00", "timestamp": 1705962550.766623}}}
{"text": "2024-01-23 06:29:10 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'LOVE', 'url': 'https://store.epicgames.com/en-US/p/love-65c4d0'}\n", "record": {"elapsed": {"repr": "0:00:08.179228", "seconds": 8.179228}, "exception": null, "extra": {"title": "LOVE", "url": "https://store.epicgames.com/en-US/p/love-65c4d0"}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3635, "name": "MainProcess"}, "thread": {"id": 140311102258048, "name": "MainThread"}, "time": {"repr": "2024-01-23 06:29:10.992632+08:00", "timestamp": 1705962550.992632}}}
{"text": "2024-01-23 06:29:15 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/love-65c4d0'}\n", "record": {"elapsed": {"repr": "0:00:13.039350", "seconds": 13.03935}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/love-65c4d0"}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3635, "name": "MainProcess"}, "thread": {"id": 140311102258048, "name": "MainThread"}, "time": {"repr": "2024-01-23 06:29:15.852754+08:00", "timestamp": 1705962555.852754}}}
{"text": "2024-01-23 06:29:25 | INFO    | epic_games.agent:claim_weekly_games:326 | claim_weekly_games - {'action': 'move to webPurchaseContainer iframe'}\n", "record": {"elapsed": {"repr": "0:00:22.394438", "seconds": 22.394438}, "exception": null, "extra": {"action": "move to webPurchaseContainer iframe"}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 326, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3635, "name": "MainProcess"}, "thread": {"id": 140311102258048, "name": "MainThread"}, "time": {"repr": "2024-01-23 06:29:25.207842+08:00", "timestamp": 1705962565.207842}}}
{"text": "2024-01-23 06:29:27 | INFO    | epic_games.agent:claim_weekly_games:328 | claim_weekly_games - {'action': 'click payment button'}\n", "record": {"elapsed": {"repr": "0:00:24.878911", "seconds": 24.878911}, "exception": null, "extra": {"action": "click payment button"}, "file": {"name": "agent.py", "path": "/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 328, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3635, "name": "MainProcess"}, "thread": {"id": 140311102258048, "name": "MainThread"}, "time": {"repr": "2024-01-23 06:29:27.692315+08:00", "timestamp": 1705962567.692315}}}
{"text": "2024-01-23 06:29:32 | ERROR    | __main__:run:158 | An error has been caught in function 'run', process 'MainProcess' (3635), thread 'MainThread' (140311102258048): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/claim.py\", line 162, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/claim.py\", line 158, in run\n    await agent.stash()\n  File \"/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/claim.py\", line 151, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/claim.py\", line 122, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 88, in async_wrapped\n    return await fn(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 47, in __call__\n    do = self.iter(retry_state=retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 314, in iter\n    return fut.result()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 50, in __call__\n    result = await fn(*args, **kwargs)\n  File \"/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/epic_games/agent.py\", line 335, in claim_weekly_games\n    await self.handle.insert_challenge(self._solver, page, wpc, payment_btn, recur_url, is_uk)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 88, in async_wrapped\n    return await fn(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 47, in __call__\n    do = self.iter(retry_state=retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 314, in iter\n    return fut.result()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/tenacity/_asyncio.py\", line 50, in __call__\n    result = await fn(*args, **kwargs)\n  File \"/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/epic_games/agent.py\", line 105, in insert_challenge\n    response = await solver.execute(window=\"free\")\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/agents/playwright/control.py\", line 626, in execute\n    self._parse_label()\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/agents/playwright/control.py\", line 240, in _parse_label\n    self.label = handle(self.prompt)\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/components/prompt_handler.py\", line 100, in handle\n    return split_prompt_message(label_cleaning(x), \"en\")\n  File \"/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/hcaptcha_challenger/components/prompt_handler.py\", line 57, in split_prompt_message\n    return re.split(r\"all (.*) images\", prompt_message)[1].strip()\nIndexError: list index out of range\n", "record": {"elapsed": {"repr": "0:00:30.171056", "seconds": 30.171056}, "exception": {"type": "IndexError", "value": "list index out of range", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/Epic-game-Free/Epic-game-Free/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 158, "message": "An error has been caught in function 'run', process 'MainProcess' (3635), thread 'MainThread' (140311102258048):", "module": "claim", "name": "__main__", "process": {"id": 3635, "name": "MainProcess"}, "thread": {"id": 140311102258048, "name": "MainThread"}, "time": {"repr": "2024-01-23 06:29:32.984460+08:00", "timestamp": 1705962572.98446}}}
